### PR TITLE
Add code-solution

### PR DESCRIPTION
**Pull request by djuks**

### CODE REVIEW

File: lib/anagrams.rb [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+    anagrams.each do |line|
+      print "#{line.each {|word| word.delete!("\n")}.join(", ")}\n"
+    end
+  end
+
+  private
+  def read(file_name)
+    Dir.chdir(File.dirname(__FILE__))
+    File.open(file_name) do |file|
+      file.readlines()
```
Comment (Ikera) [inline lib/anagrams.rb]: You don't need `()` at the end of method that doesn't have attributes.

Comment (Ikera) [inline lib/anagrams.rb]: It seems to me that the program can be triggered only with files that live under `lib` folder. Can this be changed to accept any file path as an argument?

File: lib/anagrams.rb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+    dictionary = read(file_name)
+    all_anagrams = dictionary.group_by {|word| word.downcase.delete("^a-z").each_char.sort}.values
+    result = all_anagrams.select { |element| element.length > 1 }
+    result
+  end
+
+  def print_anagrams(file_name)
+    anagrams = anagrams(file_name)
+    anagrams.each do |line|
+      print "#{line.each {|word| word.delete!("\n")}.join(", ")}\n"
```
Comment (Ikera) [inline lib/anagrams.rb]: I think that you can use `puts` instead of `print "....\n"`

File: lib/anagrams.rb [hunk 3]
```diff
@@ -0,0 +1,23 @@
+class Anagram
+  def anagrams(file_name)
+    dictionary = read(file_name)
+    all_anagrams = dictionary.group_by {|word| word.downcase.delete("^a-z").each_char.sort}.values
+    result = all_anagrams.select { |element| element.length > 1 }
+    result
+  end
```
Comment (Ikera) [inline lib/anagrams.rb]: As Slobodan said, if you can separate file reading from finding anagrams then this method won't depend on any file and testing would be much easier. Try to do that.

File: lib/anagrams.rb [hunk 4]
```diff
@@ -0,0 +1,23 @@
+class Anagram
```
Comment (Ikera) [inline lib/anagrams.rb]: In order to follow some Ruby/Rails conventions, the class name should be equal to the file name.

**Review by Ikera — COMMENTED** [review]
@Djurica-Strahinja a part of code that actually finds anagrams looks ok to me. Also, the code speed is ok too IMHO. But there are a few things that can be improved, mostly related to easier testing and ability to pass any file from the file system as an argument.

File: lib/anagrams.rb [hunk 5]
```diff
@@ -0,0 +1,20 @@
+class Anagrams
+  def find_anagrams(word_list)
+    all_anagrams = word_list.group_by {|word| word.downcase.delete("^a-z").each_char.sort}.values
```
Comment (Ikera) [inline lib/anagrams.rb]: Missing space `{ |word| ....sort }`

File: lib/anagrams.rb [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+    result = all_anagrams.select { |element| element.length > 1 }
+    result
+  end
+
+  def print_anagrams(file_name)
+    anagrams = find_anagrams(read(file_name))
+    anagrams.each do |line|
+      puts "#{line.each {|word| word.delete!("\n")}.join(", ")}\n"
```
Comment (Ikera) [inline lib/anagrams.rb]: You don't need `\n` at the end.

File: spec/anagrams_spec.rb [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+    sample = Anagrams.new
+    all_anagrams = sample.find_anagrams(["cile","mile","pile","LEMI","drakos","drako's"])
+    it 'should get all anagrams' do
+      expect(all_anagrams.count).to be 2
+    end
+
+    it 'should include test_arr' do
+      test_arr = ["mile", "LEMI"]
+      expect(all_anagrams).to include(test_arr)
+    end
```
Comment (Ikera) [inline spec/anagrams_spec.rb]: Redundant test, it's the same like one below.

File: spec/anagrams_spec.rb [hunk 8]
```diff
@@ -0,0 +1,34 @@
...
+  end
+
+  describe ".read" do
+    sample = Anagrams.new
+    file = sample.read("lib/simple-anagrams.txt")
+    it 'should get all anagrams from file' do
+      expect(sample.find_anagrams(file).count).to be 2
+    end
+  end
+end
```
Comment (Ikera) [inline spec/anagrams_spec.rb]: If you test the `read` method then you should check the output of the read method. In this case, you actually test the  `find_anagrams` method, only in a different way.

File: spec/anagrams_spec.rb [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+
+    it 'should be non-letter insensitive' do
+      test_arr = ["drakos", "drako's"]
+      expect(all_anagrams).to include(test_arr)
+    end
+  end
+
+  describe ".read" do
+    sample = Anagrams.new
+    file = sample.read("lib/simple-anagrams.txt")
```
Comment (Ikera) [inline spec/anagrams_spec.rb]: Test files should live in the `spec` directory.

**Review by Ikera — APPROVED** [review]
After you change those few small things this PR is approved from my side. 👍

**Review by basti — APPROVED** [review]

**Pull request merged**